src/physics/autompo.jl

```diff
@@ -203,6 +213,12 @@ end
 
 Base.size(ampo::OpSum) = size(data(ampo))
 
+function Base.iterate(os::OpSum, state=1)
```

Comment:
A simpler definition would be:
```julia
iterate(os::OpSum, args...) = iterate(data(os), args...)
```
i.e. just leverage the definition for `Vector`.